hypothesis-python/src/hypothesis/core.py

```diff
@@ -76,9 +76,13 @@
     from coverage.collector import FileDisposition
 
 if False:
-    from typing import Any, Dict, Callable, Optional, Union  # noqa
+    from typing import (  # noqa
+        Any, Dict, Callable, Hashable, Optional, Union, TypeVar,
+    )
     from hypothesis.utils.conventions import InferType  # noqa
 
+    T = TypeVar('T')
```

Comment:
I'd write this as `TestFunc = TypeVar('TestFunc', bound=callable)`

Reply:
Wouldn't this be over-specifying the input type? The decorators only require that the fields they are using are writable, whether or not the input is callable should be enforced by whatever consumes the output (i.e `@given`)

Reply:
It would not cause an error to pass some non-callable object to `seed` - e.g. `a = seed(0)(None)` - but it's also not a valid use of the API.  Since the whole point of type hints is to catch incorrect use, I try to be as strict as practical without ruling out valid uses.